### PR TITLE
optimize append_any more

### LINKED ISSUE #31565: apply_any gets specialized despite annotations to the contrary
After running the JuliaInterpreter tests, I get this:
```julia
julia> m = first(methods(Base.append_any))
append_any(xs...) in Base at essentials.jl:663

julia> m.specializations.func
MethodInstance for append_any(::Vararg{Any,N} where N)

julia> m.specializations.next.func
MethodInstance for append_any(::Tuple{Any}, ::Vararg{Any,N} where N)

julia> m.specializations.next.next.func
MethodInstance for append_any(::Any, ::Vararg{Any,N} where N)
```
This is despite the intention to avoid specialization: https://github.com/JuliaLang/julia/blob/b073bb5ba693565859b8e9b4179c2d8402289c1c/base/essentials.jl#L662-L663 Unfortunately, this isn't fixed by changing the declaration to
```julia
function append_any(@nospecialize xs...)
```

### PR DESCRIPTION
**Pull request by vtjnash**
Instead of having two copies of this function with different optimizations, merge them into one place (this is used by `_apply`).

note: If you were relying on this function previously (e.g. to simulate `Core._apply` in an Interpreter), I think the simplest / most direct replacement of this function is:
```julia
append_any(@nospecialize x...) = append!([], Core.svec((x...)...))
```

closes #31565

### CODE REVIEW

**Review by JeffBezanson — APPROVED** [review]
LGTM. Commits should be separated, or at least not squashed on merge.

**Comment by vtjnash** [comment]
Well, this is awkward. I selected "merge", but github UX glitched and squashed them anyways

**Review by Keno — COMMENTED** [review]
I had this review comment lying around in uncommited state. Not sure if relevant any more, but there you go.

File: src/module.c [hunk 1]
```diff
@@ -77,7 +76,6 @@ JL_DLLEXPORT void jl_set_istopmod(jl_module_t *self, uint8_t isprimary)
     self->istopmod = 1;
     if (isprimary) {
         jl_top_module = self;
-        jl_append_any_func = NULL;
```
[the remark this answers is not in the document]
Comment (vtjnash) [reply src/module.c]: It wasn't set anyways, so didn't seem to make much difference. OTOH, we could implement my suggestion in #26001 pretty quickly/easily, now taking care of this completely, and closing out your issue there.

**Pull request merged**